src/containers/Tenant/utils/schemaControls.tsx

```diff
@@ -0,0 +1,80 @@
+import {Dispatch} from 'react';
+import cn from 'bem-cn-lite';
+
+import {NavigationTreeNodeType, NavigationTreeProps} from 'ydb-ui-components';
+
+import {Button} from '@gravity-ui/uikit';
+
+import {SetQueryModeIfAvailable} from '../../../utils/hooks';
+import {setShowPreview} from '../../../store/reducers/schema/schema';
+import {setQueryTab, setTenantPage} from '../../../store/reducers/tenant/tenant';
+import {TENANT_PAGES_IDS, TENANT_QUERY_TABS_ID} from '../../../store/reducers/tenant/constants';
+import {Icon} from '../../../components/Icon';
+
+import i18n from '../i18n';
+
+import './SchemaControls.scss';
+
+interface ActionsAdditionalEffects {
+    setQueryMode: SetQueryModeIfAvailable;
+    setActivePath: (path: string) => void;
+}
+
+const bindActions = (
+    path: string,
+    dispatch: Dispatch<any>,
+    additionalEffects: ActionsAdditionalEffects,
+) => {
+    const {setActivePath} = additionalEffects;
+
+    return {
+        openPreview: () => {
+            dispatch(setShowPreview(true));
+            dispatch(setTenantPage(TENANT_PAGES_IDS.query));
+            dispatch(setQueryTab(TENANT_QUERY_TABS_ID.newQuery));
+            setActivePath(path);
+        },
+    };
+};
+
+type Controls = ReturnType<Required<NavigationTreeProps>['renderAdditionalNodeElements']>;
+
+const b = cn('show-preview-button');
+
+export const getControls =
+    (dispatch: Dispatch<any>, additionalEffects: ActionsAdditionalEffects) =>
+    (path: string, type: NavigationTreeNodeType) => {
+        const options = bindActions(path, dispatch, additionalEffects);
+        const openPreview = (
+            <Button
+                view="flat-secondary"
+                onClick={options.openPreview}
+                className={b()}
+                title={i18n('actions.openPreview')}
+                size="s"
+            >
+                <Icon name="tablePreview" height={16} width={16} />
```

Comment:
Can you use `IconWrapper` instead (it's located inside `Icon`, the issue: https://github.com/ydb-platform/ydb-embedded-ui/issues/344)? In this case you probably won't need the style
```
.show-preview-button {
    .yc-button__text {
    } 
}
```